fluent-syntax/src/errors.js

```diff
@@ -1 +1,51 @@
-export class ParseError extends Error {}
+export class ParseError extends Error {
+  constructor(code, ...args) {
+    super();
+    this.code = code;
+    this.message = getErrorMessage(code, args);
+  }
+}
+
+function getErrorMessage(code, args) {
```

Comment:
The list of errors comes from https://github.com/projectfluent/fluent-rs/blob/master/src/syntax/errors/list.rs. It's a first draft and I'm sure the list will change and expand as we start writing more tests.